source/keymap.js

```diff
@@ -23,115 +23,184 @@
 		66 : 'b'
 	});
 
-})(enyo, this);
+	enyo.singleton({
+		name: 'moon.History',
 
-enyo.singleton({
-	name: 'moon.BackKeySupport',
-	/**
-	* Target object of back key.
-	* It is also the last object which history.pushState() is executed.
-	* When we press back key or call history.back(), it is target object.
-	*
-	* @private
-	*/
-	_currentObj: undefined,
+		/**
+		* Target object of back key.
+		* It is also the last object which history.pushState() is executed.
+		* When we press back key or call history.back(), it is target object.
+		*
+		* @private
+		*/
+		_currentObj: null,
 
-	/**
-	* Stack of Back key target.
-	* This property is a plan B when history API is not allowed,
-	*
-	* @private
-	*/
-	_backKeyStack: [],
+		/**
+		* Stack of Back key target and handler
+		*
+		* @private
+		*/
+		_backHistoryStack: [],
 
-	/**
-	* history.forward() and history.go() can fire popstate event
-	* as does history.back().
-	* However we should accept popstate event only came from history.back().
-	* When it is true, we will skip popstate event.
-	*
-	* @private
-	*/
-	_ignorePopState: false,
+		/**
+		* history.forward() and history.go() can fire popstate event
+		* as does history.back().
+		* However we should accept popstate event only came from history.back().
+		* When it is true, we will skip popstate event.
+		*
+		* @private
+		*/
+		_ignorePopState: false,
 
-	/**
-	* @public
-	*/
-	getIgnorePopState: function() {
-		return this._ignorePopState;
-	},
+		/**
+		* If "disableBackHistoryAPI" in AppInfo.json is set to true, this property
+		* should be true
+		* It should be binded with AppInfo.json. Will be replaced using enyo.LunaService
+		*
+		* @type {Bollean}
+		* @default false
+		* @public
+		*/
+		disableBackHistoryAPI: false,
```

Comment:
@otisnate Ya, for better understanding of developers, I agree with your opinion.
Let me fix it when I bind it with `disableBackHistoryAPI` property in AppInfo.json